backend/app/notify/slack.go

```diff
@@ -0,0 +1,117 @@
+package notify
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	log "github.com/go-pkgz/lgr"
+	"github.com/pkg/errors"
+	"github.com/slack-go/slack"
+)
+
+// Slack implements notify.Destination for Slack
+type Slack struct {
+	channelID string
+	client    *slack.Client
+}
+
+// NewSlack makes Slack bot for notifications
+func NewSlack(token, channelName string) (*Slack, error) {
+	return newSlack(token, channelName, 0, "")
+}
+
+func newSlack(token, channelName string, _ time.Duration, apiURL string) (*Slack, error) {
```

Comment:
is there are a good reason for a separate method here? The only one I can think of is the ability to test it, but I'd rather see apiURL as a private struct filed and setting the value directly. Also, if time.Duration not used why it is even here? If this duration defines the timeout we definitely need it as we don't want to wait forever on the submission

Reply:
I have refactored accordingly. The `NewSlack` function now takes as optional arguments the slack configuration.